### ISSUE TITLE
Assert to check invariants and abort contract call

### ISSUE THREAD

**Issue by njordhov** [clarity-lang/reference#26]
Clarity should have an `assert` construct that aborts the contract call when an invariant predicate is false.

Assertions are [common](https://rosettacode.org/wiki/Assertions) in programming languages, including other Lisp dialects: Clojure has [assert](https://clojuredocs.org/clojure.core/assert); Common Lisp has [assert](http://www.lispworks.com/documentation/lw50/CLHS/Body/m_assert.htm);  Scheme has [assert](https://www.scheme.com/tspl4/exceptions.html); Typed Racket has [with-asserts](https://docs.racket-lang.org/ts-reference/Utilities.html#%28form._%28%28lib._typed-racket%2Fbase-env%2Fprims..rkt%29._with-asserts%29%29); 

Clarity already have an [`asserts!`](https://docs.blockstack.org/references/clarity-language#asserts) form, but instead of aborting the contract call, it returns an error from the current function. This is a foot gun for Clarity developers, inviting security flaws such as failing to handle severe errors. It is also inflexible, limiting assertions to functions that return a result type.

**Comment by friedger** [comment]
It could be named `asserts-panic`, or 'asserts-abort` as proposed in #25

**Comment by njordhov** [comment]
> It could be named asserts-panic, or asserts-abort

Extra verbosity not required as just `assert` is widely recognized for this functionality.

**Comment by obycode** [comment]
I would disagree with the statement that `asserts!` is a foot gun. Clarity forces users to unwrap `response` types, so users are always forced to consider the errors returned from called functions.

I do see a value in adding a `panic` variant that could be called from functions that are not already returning a `response`. Since `asserts!` is now already established, I think we would need to go with the naming suggested by Friedger for this new function, `asserts-panic`.
